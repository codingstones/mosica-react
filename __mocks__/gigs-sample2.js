export const sampleGigs = [
  {
    'day': 'Sábado, 20 de Agosto',
    'gigs': [
      {
        'title': 'rapsus',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'cesareo alierta, 5001,',
        'description': 'Concierto de rapsus',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'Avenida Cesáreo Alierta (mobilityweek)',
        'slug': 'javat',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      }
    ]
  }
]
