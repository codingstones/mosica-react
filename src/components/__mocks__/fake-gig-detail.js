export const fakeGigDetail = {
  address: "Avda. Cesáreo Alierta, 120, 50013, Zaragoza",
  affiliate_url: null,
  day: "2017-12-16",
  description:
    "Enrique Bunbury ha anunciado el lanzamiento de su nuevo disco de estudio Expectativas. Hará una parada en Zaragoza el próximo 16 de diciembre para presentarlo ante sus paisanos. Este nuevo álbum es la continuación de Palosanto de 2013. La espera ha sido de cuatro años, pero el 8 de septiembre llegarán por fin los dos primeros adelantos,",
  hour: "22:00",
  id: 6533,
  image_url: "http://www.zaragoza.es/cont/paginas/actividades/imagen/pfotoevento4020.png",
  lat_lng: "41.63602275205834,-0.8658168875758695",
  place: "Pabellón Príncipe Felipe",
  price: "",
  schedule: [],
  schedule_id: 9119,
  slug: "concierto-enrique-bunbury",
  title: "Concierto Enrique Bunbury"
}