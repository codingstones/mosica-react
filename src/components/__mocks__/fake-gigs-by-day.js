export const fakeGigsByDay = [
  {
    'day': 'Lunes, 18 de Septiembre',
    'gigs': [
      {
        'id': '123456',
        'title': 'Anarchy in the JS',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'SCBCN',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
      {
        'id': '2222222',
        'title': 'Anarchy in the Codemotion',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Another Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.wtf.es/wtf.png',
        'affiliate_url': null,
        'place': 'Around the world',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-12-03',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
    ]
  }
]

export const FIRST_DAY_GIG_TEXTS = fakeGigsByDay[0].gigs.map(gig => gig.title)
export const FIRST_GIG = fakeGigsByDay[0].gigs[0]
