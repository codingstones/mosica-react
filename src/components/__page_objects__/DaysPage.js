import toJson from 'enzyme-to-json'
import { wait } from '../../../test/helpers'
import { GigRow } from '../GigRow'

export default class DaysPage {
  constructor(wrapper) {
    this.wrapper = wrapper
  }

  async wait() {
    await wait()
    return this.wrapper.update()
  }

  matchSnapshot() {
    expect(toJson(this.wrapper)).toMatchSnapshot()
  }

  dayTexts() {
    return this.wrapper.find('.day').map((day) => day.text())
  }

  gigRowsFor(dayNumber) {
    return this.wrapper.find('.gigs').at(dayNumber).find('.row').map((row) => row.text())
  }

  containsText(text) {
    expect(this.text()).toContain(text)
  }

  text() {
    return this.wrapper.text()
  }

  html() {
    return this.wrapper.html()
  }

  setRouter(router) {
    this.wrapper.instance().router = router
  }

  async clickFirstGig() {
    return this.wrapper.find(GigRow).first().simulate('click')
  }

  clickSecondGig() {
    return this.wrapper.find(GigRow).at(1).simulate('click')
  }
}