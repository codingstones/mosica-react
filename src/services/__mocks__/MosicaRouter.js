const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MosicaRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MosicaRouter
}

