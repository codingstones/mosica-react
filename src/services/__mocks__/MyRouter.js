const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MyRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MyRouter: MyRouter,
}

