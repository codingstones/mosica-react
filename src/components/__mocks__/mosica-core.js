import { fakeGigsByDay } from './fake-gigs-by-day'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(fakeGigsByDay)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
