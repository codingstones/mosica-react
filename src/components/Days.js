import React, { Component } from 'react';
import './App.css';
import fetchJsonp from 'fetch-jsonp'
import * as mosicaCore from 'mosica-core'
import { HttpClient } from '../services/HttpClient'
import { Day } from './Day';
import { LoadSpinner } from './LoadSpinner';
import { withRouter } from 'react-router-dom'
import { MosicaRouter } from '../services/MosicaRouter';


export class DaysWithoutRouter extends Component {
  constructor(props) {
    super(props)
    this.state = { gigsByDay: [], isLoading: true }
    this.router = MosicaRouter(this.props.history)
  }

  async componentDidMount() {
    const gigService = new mosicaCore.GigService(HttpClient(fetchJsonp), new mosicaCore.Matcher())
    const gigsByDay = await gigService.retrieveNextGigs()
    this.setState({ isLoading: false })
    return this.setState({ gigsByDay })
  }

  goToGig = (gigId) => ()=> {
    this.router.navigateToGig(gigId);
  }

  render() {
    return (
      <div>
        <LoadSpinner isLoading={this.state.isLoading}/>
        {this.state.gigsByDay.map((day) =>
          <Day key={day.day} day={day} onClick={this.goToGig}/>
        )}
      </div>
    );
  }
}

export default withRouter(DaysWithoutRouter);
