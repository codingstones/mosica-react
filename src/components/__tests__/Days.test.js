import React from 'react'
import { fakeGigsByDay,FIRST_DAY_GIG_TEXTS } from '../__mocks__/fake-gigs-by-day'
import DaysPage from '../__page_objects__/DaysPage'
import { DaysWithoutRouter as Days } from '../Days'
import { Wrap } from '../../../test/helpers'
//We don't need jest.mock(..MosicaCore) cause it's a default manual mock

describe('Days', async () => {

  let page
  beforeEach(async () => {
    const wrapper = Wrap(<Days/>).mount()
    page = new DaysPage(wrapper)
    await page.wait()
  })

  it('renders the texts of the first day of gigs', async () => {
    FIRST_DAY_GIG_TEXTS.map((text) =>
      page.containsText(text)
    )
  })

  it('full snapshot', () => {
    page.matchSnapshot()
  })

  describe('When clicking gig', ()=> {
    let routerSpy
    beforeEach(async () => {
      routerSpy = { navigateToGig: jest.fn() }
      page.setRouter(routerSpy)
    })

    it('navigates to first gig detail', async () => {
      const FIRST_GIG_ID = '123456'
      page.clickFirstGig()
      expect(routerSpy.navigateToGig).toHaveBeenCalledWith(FIRST_GIG_ID)
    })

    it('navigates to second gig detail', async () => {
      const SECOND_GIG_ID = '2222222'
      page.clickSecondGig()
      expect(routerSpy.navigateToGig).toHaveBeenCalledWith(SECOND_GIG_ID)
    })
  })

  it('shows gigs for each day', async () => {
    //IS THIS TEST WORTHY? Maybe easier to unit-test components
    let expectedGigs
    fakeGigsByDay.map((day, index)=> {
      expectedGigs = day.gigs.map((gig) => gig.title + '-' + gig.place)
      expect(page.gigRowsFor(index)).toEqual(expectedGigs)
    })
  })

})

