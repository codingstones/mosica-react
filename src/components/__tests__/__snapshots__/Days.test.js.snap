// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Days full snapshot 1`] = `
<DaysWithoutRouter>
  <div>
    <Component
      isLoading={false}
    />
    <Component
      day={
        Object {
          "day": "Lunes, 18 de Septiembre",
          "gigs": Array [
            Object {
              "address": "BCN",
              "affiliate_url": null,
              "day": "2017-09-18",
              "description": "Coding Stones GigRow",
              "hour": "19:30",
              "id": "123456",
              "image_url": "http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png",
              "lat_lng": "41.641851935961654,-0.8751129897638315",
              "place": "SCBCN",
              "price": "",
              "schedule": Array [
                Object {
                  "created_at": "2017-09-07T15:57:35.419Z",
                  "day": "2017-09-18",
                  "gig_id": 5927,
                  "hour": "19:30",
                  "id": 5280,
                  "updated_at": "2017-09-07T15:57:35.419Z",
                },
              ],
              "schedule_id": 5280,
              "slug": "stones",
              "title": "Anarchy in the JS",
            },
            Object {
              "address": "BCN",
              "affiliate_url": null,
              "day": "2017-12-03",
              "description": "Another Coding Stones GigRow",
              "hour": "19:30",
              "id": "2222222",
              "image_url": "http://www.wtf.es/wtf.png",
              "lat_lng": "41.641851935961654,-0.8751129897638315",
              "place": "Around the world",
              "price": "",
              "schedule": Array [
                Object {
                  "created_at": "2017-09-07T15:57:35.419Z",
                  "day": "2017-09-18",
                  "gig_id": 5927,
                  "hour": "19:30",
                  "id": 5280,
                  "updated_at": "2017-09-07T15:57:35.419Z",
                },
              ],
              "schedule_id": 5280,
              "slug": "stones",
              "title": "Anarchy in the Codemotion",
            },
          ],
        }
      }
      key="Lunes, 18 de Septiembre"
      onClick={[Function]}
    >
      <div>
        <Header
          as="h2"
          textAlign="center"
        >
          <h2
            className="ui center aligned header"
          >
            Lunes, 18 de Septiembre
          </h2>
        </Header>
        <SegmentGroup
          raised={true}
        >
          <div
            className="ui raised segments"
          >
            <ItemGroup
              divided={true}
              key="Lunes, 18 de Septiembre"
              link={true}
            >
              <div
                className="ui divided link items"
              >
                <Component
                  gig={
                    Object {
                      "address": "BCN",
                      "affiliate_url": null,
                      "day": "2017-09-18",
                      "description": "Coding Stones GigRow",
                      "hour": "19:30",
                      "id": "123456",
                      "image_url": "http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png",
                      "lat_lng": "41.641851935961654,-0.8751129897638315",
                      "place": "SCBCN",
                      "price": "",
                      "schedule": Array [
                        Object {
                          "created_at": "2017-09-07T15:57:35.419Z",
                          "day": "2017-09-18",
                          "gig_id": 5927,
                          "hour": "19:30",
                          "id": 5280,
                          "updated_at": "2017-09-07T15:57:35.419Z",
                        },
                      ],
                      "schedule_id": 5280,
                      "slug": "stones",
                      "title": "Anarchy in the JS",
                    }
                  }
                  key="123456"
                  onClick={[Function]}
                >
                  <Item
                    key="123456"
                    onClick={[Function]}
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <ItemContent>
                        <div
                          className="content"
                        >
                          <ItemHeader
                            as="h3"
                          >
                            <h3
                              className="header"
                            >
                              Anarchy in the JS
                            </h3>
                          </ItemHeader>
                          <ItemDescription>
                            <div
                              className="description"
                            >
                              <Place
                                place="SCBCN"
                              >
                                <span>
                                  <Icon
                                    as="i"
                                    color="red"
                                    name="marker"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="red marker icon"
                                    />
                                  </Icon>
                                  SCBCN
                                </span>
                              </Place>
                              <span>
                                <Time
                                  hour="19:30"
                                >
                                  <span>
                                    <Icon
                                      as="i"
                                      color="green"
                                      name="time"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="green time icon"
                                      />
                                    </Icon>
                                    19:30
                                     horas
                                  </span>
                                </Time>
                              </span>
                            </div>
                          </ItemDescription>
                          <ItemExtra>
                            <div
                              className="extra"
                            >
                              <Price
                                price=""
                              />
                              <span />
                              <Likes>
                                <div>
                                  <Button
                                    as="button"
                                    color="red"
                                    content="Like"
                                    icon="heart"
                                    label={
                                      Object {
                                        "basic": true,
                                        "color": "red",
                                        "content": 160,
                                        "pointing": "left",
                                      }
                                    }
                                    onClick={[Function]}
                                    size="mini"
                                  >
                                    <div
                                      className="ui labeled button"
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="ui red mini button"
                                        tabIndex={0}
                                      >
                                        <Icon
                                          as="i"
                                          key="heart"
                                          name="heart"
                                        >
                                          <i
                                            aria-hidden="true"
                                            className="heart icon"
                                          />
                                        </Icon>
                                         
                                        Like
                                      </button>
                                      <Label
                                        basic={true}
                                        color="red"
                                        content={160}
                                        pointing="left"
                                      >
                                        <div
                                          className="ui red left pointing basic label"
                                          onClick={[Function]}
                                        >
                                          160
                                        </div>
                                      </Label>
                                    </div>
                                  </Button>
                                </div>
                              </Likes>
                              <BuyTickets
                                url="any"
                              >
                                <Button
                                  as="button"
                                  floated="right"
                                  onClick={[Function]}
                                  primary={true}
                                >
                                  <button
                                    className="ui primary right floated button"
                                    onClick={[Function]}
                                    role="button"
                                  >
                                    Buy tickets
                                    <Icon
                                      as="i"
                                      name="right chevron"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="right chevron icon"
                                      />
                                    </Icon>
                                  </button>
                                </Button>
                              </BuyTickets>
                            </div>
                          </ItemExtra>
                        </div>
                      </ItemContent>
                    </div>
                  </Item>
                </Component>
                <Component
                  gig={
                    Object {
                      "address": "BCN",
                      "affiliate_url": null,
                      "day": "2017-12-03",
                      "description": "Another Coding Stones GigRow",
                      "hour": "19:30",
                      "id": "2222222",
                      "image_url": "http://www.wtf.es/wtf.png",
                      "lat_lng": "41.641851935961654,-0.8751129897638315",
                      "place": "Around the world",
                      "price": "",
                      "schedule": Array [
                        Object {
                          "created_at": "2017-09-07T15:57:35.419Z",
                          "day": "2017-09-18",
                          "gig_id": 5927,
                          "hour": "19:30",
                          "id": 5280,
                          "updated_at": "2017-09-07T15:57:35.419Z",
                        },
                      ],
                      "schedule_id": 5280,
                      "slug": "stones",
                      "title": "Anarchy in the Codemotion",
                    }
                  }
                  key="2222222"
                  onClick={[Function]}
                >
                  <Item
                    key="2222222"
                    onClick={[Function]}
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <ItemContent>
                        <div
                          className="content"
                        >
                          <ItemHeader
                            as="h3"
                          >
                            <h3
                              className="header"
                            >
                              Anarchy in the Codemotion
                            </h3>
                          </ItemHeader>
                          <ItemDescription>
                            <div
                              className="description"
                            >
                              <Place
                                place="Around the world"
                              >
                                <span>
                                  <Icon
                                    as="i"
                                    color="red"
                                    name="marker"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="red marker icon"
                                    />
                                  </Icon>
                                  Around the world
                                </span>
                              </Place>
                              <span>
                                <Time
                                  hour="19:30"
                                >
                                  <span>
                                    <Icon
                                      as="i"
                                      color="green"
                                      name="time"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="green time icon"
                                      />
                                    </Icon>
                                    19:30
                                     horas
                                  </span>
                                </Time>
                              </span>
                            </div>
                          </ItemDescription>
                          <ItemExtra>
                            <div
                              className="extra"
                            >
                              <Price
                                price=""
                              />
                              <span />
                              <Likes>
                                <div>
                                  <Button
                                    as="button"
                                    color="red"
                                    content="Like"
                                    icon="heart"
                                    label={
                                      Object {
                                        "basic": true,
                                        "color": "red",
                                        "content": 236,
                                        "pointing": "left",
                                      }
                                    }
                                    onClick={[Function]}
                                    size="mini"
                                  >
                                    <div
                                      className="ui labeled button"
                                      onClick={[Function]}
                                    >
                                      <button
                                        className="ui red mini button"
                                        tabIndex={0}
                                      >
                                        <Icon
                                          as="i"
                                          key="heart"
                                          name="heart"
                                        >
                                          <i
                                            aria-hidden="true"
                                            className="heart icon"
                                          />
                                        </Icon>
                                         
                                        Like
                                      </button>
                                      <Label
                                        basic={true}
                                        color="red"
                                        content={236}
                                        pointing="left"
                                      >
                                        <div
                                          className="ui red left pointing basic label"
                                          onClick={[Function]}
                                        >
                                          236
                                        </div>
                                      </Label>
                                    </div>
                                  </Button>
                                </div>
                              </Likes>
                              <BuyTickets
                                url="any"
                              >
                                <Button
                                  as="button"
                                  floated="right"
                                  onClick={[Function]}
                                  primary={true}
                                >
                                  <button
                                    className="ui primary right floated button"
                                    onClick={[Function]}
                                    role="button"
                                  >
                                    Buy tickets
                                    <Icon
                                      as="i"
                                      name="right chevron"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="right chevron icon"
                                      />
                                    </Icon>
                                  </button>
                                </Button>
                              </BuyTickets>
                            </div>
                          </ItemExtra>
                        </div>
                      </ItemContent>
                    </div>
                  </Item>
                </Component>
              </div>
            </ItemGroup>
          </div>
        </SegmentGroup>
      </div>
    </Component>
  </div>
</DaysWithoutRouter>
`;
